### ISSUE TITLE
Do not use limit and offset when fetching filtered content

### ISSUE THREAD

**Issue by laurentdavid** [call-learning/moodle-local_resourcelibrary#3]
The display of the results is really inspired from the block_myoverview. However there is something rather complicated that is done in this block's javascript: because we use limit and offset to fetch only a part of the results and then we filter the list down to the "really" visible courses or module, there are times where we either: 

- return less courses/modules than expected: the page has got less modules than expected AND will still have a pagination to a next page for more module (in this case this should be the last page)
- we would have to fetch more course/module than the expected window size to make sure we have enough courses/results expected (actually from my understanding we would fetch twice the number of courses so to be sure). But even this would not make sure we would fill up the page.

Basically, there are good chances that the offset/limit would be not in sync with the returned list. And it gets complicated to manage this in a reliable way (it does also increase the complexity of the code).
If the aim was not to hammer the server with SQL queries, maybe we should consider instead:

- Creating a temporary tables (create_temp_table) and fill it in with a cron (for example with all courses and their current field value, so then we would just have to filter on this aggregated table)
- Use a cache
- both (?)

I guess the issue that was to be overcome in the block_overview was for the first query (where the limit was trying to cap the time spent by  the query on the first time).
